peekaboo/db.py

```diff
@@ -156,13 +159,12 @@ def analysis_save(self, sample):
 
         @param sample: The sample object for this analysis task.
         """
-        sample_info = self.sample_info_fetch(sample)
-        if sample_info is None:
```

Comment:
If this is going to be the change removing the existance constraint, it warrants an explanation as to the why in the commit message.